self_test/generic/complex_matrices_test/cr_complex_matrix_test.cc

```diff
@@ -0,0 +1,209 @@
+  rhs[2] = complex<double>(2, 1);
+  rhs[3] = complex<double>(-3, 1);
+
+  Vector<std::complex<double>> soln(4);
+
+  Vector<double> value_double(5);
+  value_double[0] = double(1);
+  value_double[1] = double(2);
+  value_double[2] = double(3);
+  value_double[3] = double(4);
+  value_double[4] = double(5);
+
+  unsigned* method_ptr = new unsigned;
+
+  // test default constructor
+  CRComplexMatrix matrix_default();
+
+  // test full matrix constructor
+  CRComplexMatrix matrix(values, column_index, row_start, n, m);
+  cout << matrix.nrow() << endl;
+  cout << matrix.ncol() << endl;
+  print_cr_complex_matrix(matrix);
+
+  CRComplexMatrix matrix_square(
+    values_square, column_index_square, row_start_square, 4, 4);
+  cout << matrix_square.nrow() << endl;
+  cout << matrix_square.ncol() << endl;
+  print_cr_complex_matrix(matrix_square);
+
+  CRComplexMatrix matrix_result(
+    values_square, column_index_square, row_start_square, 4, 4);
+  print_cr_complex_matrix(matrix_result);
+
+  unsigned ncol_double = matrix_square.ncol();
+  LinearAlgebraDistribution* dist_ptr = new LinearAlgebraDistribution();
+  CRDoubleMatrix matrix_in_double(
+    dist_ptr, ncol_double, value_double, column_index, row_start);
+
+  CRComplexMatrix matrix_square_2(
+    values_square, column_index_square, row_start_square, 4, 4);
+
+  // test LU decomposition
+  cout << matrix_square.ludecompose() << endl;
+
+  // test residual
+  matrix_square.lubksub(rhs);
+  print_complex_vector(rhs);
+
+  // test multiply
+  matrix_square.multiply(x, soln);
+  print_complex_vector(soln);
+
+  // test multiply transposed
+  matrix_square.multiply_transpose(x, soln);
+  print_complex_vector(soln);
+
+
+  // test add (CRDoubleMatrix)
+  matrix_square.add(matrix_in_double, matrix_square);
+  print_cr_complex_matrix(matrix_square);
+
+  // test add (CRComplexMatrix)
+  matrix_square.add(matrix_square, matrix_square);
+  print_cr_complex_matrix(matrix_square);
+
+  // Print input matrices
+  cout << "A" << endl;
+  print_cr_complex_matrix(matrix_square);
+  cout << "B" << endl;
+  print_cr_complex_matrix(matrix_square_2);
+
+  // test multiply method 2 (default)
+  matrix_square.multiply(matrix_square_2, matrix_result);
+  cout << "A B" << endl;
+  print_cr_complex_matrix(matrix_result);
+
+  // test multiply set method and default value
+  method_ptr = &matrix_square.serial_matrix_matrix_multiply_method();
+  cout << *method_ptr << endl;
+
+  // test multiply method 1
+  *method_ptr = 1;
+  matrix_square.multiply(matrix_square_2, matrix_result);
+  cout << *method_ptr << endl;
+  cout << "A B" << endl;
+  print_cr_complex_matrix(matrix_result);
+
+  // test multiply method 3
+  *method_ptr = 3;
+  matrix_square.multiply(matrix_square_2, matrix_result);
+  cout << *method_ptr << endl;
+  cout << "A B" << endl;
+  print_cr_complex_matrix(matrix_result);
+
+  // Print cleanup messages for easier debugging
+  cout << "Begin cleanup" << endl;
+  delete dist_ptr;
+  dist_ptr = 0;
```

Comment:
Set to `nullptr` (C++11) instead of `0` (old-C++). Ditto for other deleted pointers.